app/attributes/attributedata.h

```diff
@@ -58,6 +60,15 @@ class  FormItem
       const QgsExpression &visibilityExpression
     );
 
+    //! Contructor for relation items
```

Comment:
I think it is better to have just one constructor with ALL const items and then static factory functions 
```
static FormItem createFieldItem(...)
static FormItem createRelationItem(...)
```
with subset of attributes